lib/core/buffer.cpp

```diff
@@ -210,7 +208,7 @@ Buffer::Buffer(int num_frames, size_t len, std::shared_ptr<metadataPool> pool,
                bool zero_new_frames) :
     GenericBuffer(_buffer_name, _buffer_type, pool, num_frames), frame_size(len),
     // By default don't zero buffers at the end of their use.
-    _zero_frames(false), frames(num_frames, nullptr), frames_desc(num_frames, nullptr),
+    _zero_frames(false), frames(num_frames, nullptr), frames_desc(nullptr),
```

Comment:
If frames_desc is tailored to frames containing NDArrays, can we make this a part of an NDArray buffer type, rather than GenericBuffer?

Reply:
buffers get created before the stages that use them, so the information would then have to be present in the yaml files for kotekan to use it. This was discussed and rejected during the last software call, so this should address the question here. 

Everything in buffers can be an NDArray. Might a one-dim one with type `uint8`, but that is still an NDArray. :-)

Right now the frames are:

```
    /// The array of frames (the actual data we are carrying)
    std::vector<uint8_t*> frames;
```

pointing to the pool of frames. In an actual `NDArray` (with data, which is not what we want, since it would be one `NDArray` per frame again) this would be the 

```
    // Pointer to the array data
    T* m_data;
```

member for `T = uint8_t`.

Reply:
> Everything in buffers can be an NDArray. Might a one-dim one with type uint8, but that is still an NDArray. :-)

Well, sure, but then the GenericNDArray fields are of limited use, i.e. further structural data will remain in metadata for vis/N2/hfb frame(view)s. Future task for handling that.